### PR TITLE
Some states are missing at init in tahoma's response

### PR DESCRIPTION
**Pull request by vlebourl**
Missing states must be dynamically added later.

requires: https://github.com/iMicknl/python-tahoma-api/pull/29

### CODE REVIEW

File: custom_components/tahoma/coordinator.py [hunk 1]
```diff
@@ -52,12 +53,15 @@ async def _async_update_data(self) -> Dict[str, Device]:
                 if event.name == "DeviceStateChangedEvent":
                     for state in event.device_states:
                         device = self.devices[event.deviceurl]
-                        device.states[state.name].value = self._get_state(state)
+                        if device.states[state.name] is None:
```
Comment (tetienne) [inline custom_components/tahoma/coordinator.py:56]:
```suggestion
                        if state.name not in device.states:
```

File: custom_components/tahoma/coordinator.py [hunk 2]
```diff
@@ -52,12 +53,15 @@ async def _async_update_data(self) -> Dict[str, Device]:
                 if event.name == "DeviceStateChangedEvent":
                     for state in event.device_states:
                         device = self.devices[event.deviceurl]
-                        device.states[state.name].value = self._get_state(state)
+                        if state.name not in device.states:
```
Comment (tetienne) [inline custom_components/tahoma/coordinator.py:56]:
```suggestion
                        if state.name not in device.states:
```

File: custom_components/tahoma/coordinator.py [hunk 3]
```diff
@@ -52,12 +53,15 @@ async def _async_update_data(self) -> Dict[str, Device]:
...
+                            device.states[state.name] = state
+                        else:
```
Comment (tetienne) [inline custom_components/tahoma/coordinator.py:58]: You can remove the else. This way with the next line we will cast the state value.

File: custom_components/tahoma/climate_st.py [hunk 4]
```diff
@@ -85,9 +87,12 @@ def __init__(self, device_url: str, coordinator: TahomaDataUpdateCoordinator):
...
         if self.hvac_mode == HVAC_MODE_AUTO:
-            self._saved_target_temp = self.select_state(
-                PRESET_TEMPERATURES[self.preset_mode]
-            )
+            if self.preset_mode == PRESET_NONE:
+                self._saved_target_temp = None
+            else:
+                self._saved_target_temp = self.select_state(
+                    PRESET_TEMPERATURES[self.preset_mode]
+                )
```
Comment (tetienne) [inline custom_components/tahoma/climate_st.py:95]: Can you please what it means graphically?

File: custom_components/tahoma/climate_st.py [hunk 5]
```diff
@@ -160,6 +165,15 @@ def hvac_modes(self) -> List[str]:
         """Return the list of available hvac operation modes."""
         return [HVAC_MODE_AUTO, HVAC_MODE_HEAT]
 
+    @property
+    def hvac_action(self) -> str:
```
Comment (tetienne) [inline custom_components/tahoma/climate_st.py:169]: Cool mode is not supported?

Comment (vlebourl) [reply custom_components/tahoma/climate_st.py:169]: no. this is a heater, not an AC

File: custom_components/tahoma/climate_st.py (hunk 4 above)
Comment (vlebourl) [reply custom_components/tahoma/climate_st.py:95]: If target is None, the target temp just doesn't show on the card, and the slider stays at the bottom left end.

**Comment by iMicknl** [comment]
@vlebourl could you 'resolve conversations' when you refactored based on review? (and leave them open in case there is an ongoing discussion). Makes it easier to review, after a previous review :-)

I am ok by merging this one to @tetienne's branch, where we can review again and merge to master.

**Pull request merged**